homeassistant/components/incharge/sensor.py

```diff
@@ -0,0 +1,136 @@
+"""Get status for all stations for InCharge."""
+from __future__ import annotations
+
+import datetime
+import logging
+
+import async_timeout
+import requests
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+SENSOR_TYPES: dict[str, SensorEntityDescription] = {
+    "total_energy_consumption": SensorEntityDescription(
+        key="total_energy_consumption",
+        name="Total Energy Consumption",
+        suggested_display_precision=2,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the InCharge sensor."""
+    # Add entities for each charging station
+    incharge_data_coordinator = hass.data.setdefault(DOMAIN, {})[config_entry.entry_id]
+    response = await hass.async_add_executor_job(
+        incharge_data_coordinator.incharge_api.get_stations
+    )
+    for station in response.json()["stations"]:
+        entities = [
+            InChargeEntity(
+                coordinator=incharge_data_coordinator,
+                station_id=station["name"],
+                unique_id=f"{station['name']}-{description.name}",
+                description=description,
+            )
+            for _, description in SENSOR_TYPES.items()
+        ]
+        async_add_entities(entities, update_before_add=True)
+
+
+class InChargeEntity(CoordinatorEntity, SensorEntity):
```

Comment:
CoordinatorEntity[InChargeDataCoordinator]

Reply:
Here's an example:

https://github.com/home-assistant/core/blob/cb578c71e03e0187b5fc6a516e5d9ec11421a8e1/homeassistant/components/wled/models.py#L10

you are essentially saying what type of coordinator this coordinator entity will use.

Also typically when you name the class Integration + Entity, that is a base Entity object, and then all of your other entites are subclasses of that entity.

i.e. INChargeSensor(SensorEntity, InChargeEntity)

You don't have to do that, but you should probably name this InChargeSensor.